crunchers/cruncher_active_users.h

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+ The MIT License (MIT)
+
+ Copyright (c) 2016 Grigory Nikolaenko <nikolaenko.grigory@gmail.com>
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ *******************************************************************************/
+
+#ifndef CRUNCHER_ACTIVE_USERS_H
+#define CRUNCHER_ACTIVE_USERS_H
+
+#include <list>
+#include <unordered_map>
+
+#include "cruncher.h"
+#include "schema.h"
+
+namespace CTFO {
+
+template <typename NAMESPACE>
+struct ActiveUsersCruncherImpl {
+  using entry_t = typename NAMESPACE::CTFOLogEntry;
+  using EventLogEntry = typename NAMESPACE::EventLogEntry;
+  using iOSBaseEvent = typename NAMESPACE::iOSBaseEvent;
+
+  ActiveUsersCruncherImpl(std::chrono::microseconds interval) : interval_(interval), stream_size_(0) {}
+  virtual ~ActiveUsersCruncherImpl() = default;
+
+  void OnEvent(const entry_t& e, idxts_t idxts) {
+    current_us_ = idxts.us;
+    ++stream_size_;
+    if (Exists<EventLogEntry>(e)) {
+      OnEventLogEntry(Value<EventLogEntry>(e));
+    }
+    while (!users_list_.empty() && users_list_.back().us + interval_ <= current_us_) {
+      users_map_.erase(users_list_.back().device_id);
+      users_list_.pop_back();
+    }
+  }
+
+  uint64_t Count() const { return users_list_.size(); }
```

Comment:
`OnEvent` method above and `Count` are supposed to be called from different threads, right? What about synchronization primitives?